src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,111 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int START_CAPACITY = 10;
+    private Object[] array;
+    private int size;
+
+    public ArrayList() {
+        array = new Object[START_CAPACITY];
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == array.length) {
+            extendArray();
+        }
+        array[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkIndex(index);
+        if (size == array.length) {
+            extendArray();
+        }
+        System.arraycopy(array, index, array, index + 1, size - index);
+        array[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkBound(index);
+        return (T) array[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkBound(index);
+        array[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkBound(index);
+        T deletedElement = (T) array[index];
+        if (index != array.length - 1) {
+            System.arraycopy(array, index + 1, array, index, size - index);
+        }
+        array[--size] = null;
+        return deletedElement;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        int index = getIndexByValue(element);
+        if (index == -1) {
+            throw new NoSuchElementException("There is no such element in the list");
```

Comment:
it's better to add element to message, "There is no such element in the list" > "Element" + element + "doesn't exist" (for example)